Bottom: 76e0788f5591da6232971038fc3fe9b3c9e6e177
Top:    80406a60c91f52e62440135a6d332934b21c7f75
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-11-23 15:32:47 +0000

Disable DRC checks on "outline" and "route" layers


---

diff --git a/src/find.c b/src/find.c
index 5d2e92c..591df39 100644
--- a/src/find.c
+++ b/src/find.c
@@ -822,6 +822,9 @@ LookupLOConnectionsToPVList (bool AndRats)
       /* now all lines, arcs and polygons of the several layers */
       for (layer = 0; layer < max_copper_layer; layer++)
         {
+          if (is_layer_mechanical (LAYER_PTR (layer)))
+            continue;
+
           info.layer = layer;
           /* add touching lines */
           if (setjmp (info.env) == 0)
@@ -3977,6 +3980,9 @@ DRCAll (void)
     {
       COPPERLINE_LOOP (PCB->Data);
       {
+        if (is_layer_mechanical (layer))
+          continue;
+
         /* check line clearances in polygons */
         PlowsPolygon (PCB->Data, LINE_TYPE, layer, line, drc_callback);
         if (IsBad)
@@ -4022,6 +4028,8 @@ DRCAll (void)
     {
       COPPERARC_LOOP (PCB->Data);
       {
+        if (is_layer_mechanical (layer))
+          continue;
         PlowsPolygon (PCB->Data, ARC_TYPE, layer, arc, drc_callback);
         if (IsBad)
           break;
diff --git a/src/misc.c b/src/misc.c
index 705f8df..c43c185 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -2302,6 +2302,14 @@ ElementOrientation (ElementType *e)
   return dy > 0 ? 3 : 1;
 }
 
+bool is_layer_mechanical (LayerType *layer)
+{
+  if (strcmp (layer->Name, "outline") == 0 ||
+      strcmp (layer->Name, "route") == 0)
+    return true;
+  return false;
+}
+
 int
 ActionListRotations(int argc, char **argv, int x, int y)
 {
diff --git a/src/misc.h b/src/misc.h
index b46cad4..4338c69 100644
--- a/src/misc.h
+++ b/src/misc.h
@@ -126,6 +126,8 @@ char * GetInfoString (void);
    comparing two similar footprints.  */
 int ElementOrientation (ElementType *e);
 
+bool is_layer_mechanical (LayerType *layer);
+
 /* These are in netlist.c */
 
 void NetlistChanged (int force_unfreeze);
